drivers/sensor/lis2de12/lis2de12.c

```diff
@@ -0,0 +1,479 @@
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <string.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+
+#include "lis2de12.h"
+
+LOG_MODULE_REGISTER(LIS2DE12, CONFIG_SENSOR_LOG_LEVEL);
+
+static const float lis2de12_odr_map[10] = {
+			0.0f, 1.0f, 10.0f, 25.0f, 50.0f, 100.0f, 200.0f,
+			400.0f, 1620.0f, 5376.0f};
+
+static int lis2de12_freq_to_odr_val(const struct device *dev, uint16_t freq)
+{
+	size_t i;
+
+	for (i = 0; i < ARRAY_SIZE(lis2de12_odr_map); i++) {
+		if (freq <= lis2de12_odr_map[i]) {
+			return i;
+		}
+	}
+
+	return -EINVAL;
+}
+
+typedef struct {
+	uint16_t fs;
+	uint32_t gain; /* Accel sensor sensitivity in ug/LSB */
+} fs_map;
+
+static const fs_map lis2de12_accel_fs_map[] = {
+				{2, 15600},
+				{4, 31200},
+				{8, 62500},
+				{16, 187500}
+			};
+
+static int lis2de12_accel_range_to_fs_val(int32_t range)
+{
+	size_t i;
+
+	for (i = 0; i < ARRAY_SIZE(lis2de12_accel_fs_map); i++) {
+		if (range == lis2de12_accel_fs_map[i].fs) {
+			return i;
+		}
+	}
+
+	return -EINVAL;
+}
+
+static int lis2de12_accel_set_fs_raw(const struct device *dev, uint8_t fs)
+{
+	const struct lis2de12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	struct lis2de12_data *data = dev->data;
+
+	if (lis2de12_full_scale_set(ctx, fs) < 0) {
+		return -EIO;
+	}
+
+	data->accel_fs = fs;
+
+	return 0;
+}
+
+static int lis2de12_accel_set_odr_raw(const struct device *dev, uint8_t odr)
+{
+	const struct lis2de12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	struct lis2de12_data *data = dev->data;
+
+	if (lis2de12_data_rate_set(ctx, odr) < 0) {
+		return -EIO;
+	}
+
+	data->accel_freq = odr;
+
+	return 0;
+}
+
+static int lis2de12_accel_odr_set(const struct device *dev, uint16_t freq)
+{
+	int odr;
+
+	odr = lis2de12_freq_to_odr_val(dev, freq);
+	if (odr < 0) {
+		return odr;
+	}
+
+	if (lis2de12_accel_set_odr_raw(dev, odr) < 0) {
+		LOG_ERR("failed to set accelerometer sampling rate");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int lis2de12_accel_range_set(const struct device *dev, int32_t range)
+{
+	int fs;
+	struct lis2de12_data *data = dev->data;
+
+	fs = lis2de12_accel_range_to_fs_val(range);
+	if (fs < 0) {
+		return fs;
+	}
+
+	if (lis2de12_accel_set_fs_raw(dev, fs) < 0) {
+		LOG_ERR("failed to set accelerometer full-scale");
+		return -EIO;
+	}
+
+	data->acc_gain = lis2de12_accel_fs_map[fs].gain;
+	return 0;
+}
+
+static int lis2de12_accel_config(const struct device *dev,
+				 enum sensor_channel chan,
+				 enum sensor_attribute attr,
+				 const struct sensor_value *val)
+{
+	switch (attr) {
+	case SENSOR_ATTR_FULL_SCALE:
+		return lis2de12_accel_range_set(dev, sensor_ms2_to_g(val));
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		return lis2de12_accel_odr_set(dev, val->val1);
+	default:
+		LOG_WRN("Accel attribute %d not supported.", attr);
+		return -ENOTSUP;
+	}
+
+	return 0;
```

Comment:
I think you can drop this and the compiler should be happy since you have a return on all switch paths